### ISSUE TITLE
Lots of output on applications debug console

### ISSUE THREAD

**Issue by AndreSomers** [KDAB/GammaRay#480]
After attaching GammaRay to my application, I see quite a bit of output on the debug console:

```
TimerModel::preSignalActivate(): Recursive timeout for timer 00000267B179EFC0!
TimerModel::preSignalActivate(): Recursive timeout for timer 00000267B179EFC0!
TimerModel::preSignalActivate(): Recursive timeout for timer 00000267B179EFC0!
QEventDispatcherWin32::remainingTime: timer id 4 not found
TimerModel::preSignalActivate(): Recursive timeout for timer 00000267B179EFC0!
TimerModel::preSignalActivate(): Recursive timeout for timer 00000267B179EFC0!
QEventDispatcherWin32::remainingTime: timer id 2 not found
QEventDispatcherWin32::remainingTime: timer id 2 not found
TimerModel::preSignalActivate(): Recursive timeout for timer 00000267B179EFC0!
TimerModel::preSignalActivate(): Recursive timeout for timer 00000267B179EFC0!
QEventDispatcherWin32::remainingTime: timer id 2 not found
QEventDispatcherWin32::remainingTime: timer id 8 not found
TimerModel::preSignalActivate(): Recursive timeout for timer 00000267B179EFC0!
QEventDispatcherWin32::remainingTime: timer id 4 not found
QEventDispatcherWin32::remainingTime: timer id 7 not found
TimerModel::preSignalActivate(): Recursive timeout for timer 00000267B179EFC0!
TimerModel::preSignalActivate(): Recursive timeout for timer 00000267B179EFC0!`

The TimerModel one occurs especially frequently, but I also see other messages such as:
`QPaintDevice::metrics: Device has no metric information
QPaintDevice::metrics: Device has no metric information
```

I'd prefer it if GammaRay would not just inject its debug output into the output of the application itself.

**Comment by vkrause** [comment]
TimerModel is us, the rest is coming from Qt. @pasnox any idea about the recursive timeout warning?

**Comment by pasnox** [comment]
I can disable the warning, though in all my test case it never happen much than 2 or 3 timer per tested app.
Can you export the behavior in a test case so i can try to see what happen?

**Comment by AndreSomers** [comment]
Attach it to the current KDABViewer and start playing with the Scheduling views...

**Comment by MDoennebrink** [comment]
I have the same problem as AndreSomers.

In my application there is a communication thread with a QTimer.
The QTimer calls every 10 ms a function to process the data.
Every 10 ms the output "Recursive timeout" appears:

TimerModel::preSignalActivate(): Recursive timeout for timer 0x7f87b0003190!
TimerModel::preSignalActivate(): Recursive timeout for timer 0x7f87b0003190!
TimerModel::preSignalActivate(): Recursive timeout for timer 0x7f87b0003190!
...

I have created a small sample project that shows the error:
https://github.com/MDoennebrink/GammaRayTimeout

Used versions:
Qt 5.7.1
GammaRay v2.11.1

**Comment by godboutj** [comment]
I have the same issue, I got a lot of those trace beginning to appear once I attach Gammaray. Before the attach, no such trace. Make it super hard to follow the console output, since this get print once every few ms. It flood it. I'm on a big client project, the code is not super clean and might be due to some recursivity. Any way to disable this. or some feature into gammary can be turn off to prevent this?

**Comment by Waqar144** [comment]
@godboutj Can you try GammaRay from master and see if the issue is reproducible? I can't reproduce it
